---
title: "Supplementary Figure 1 Quality Control"
author: "Nick Hirschm√ºller"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    code_folding: hide
---

```{r setup, echo=F, message=F, error=F}
knitr::opts_chunk$set(echo = T, message = F, warning = F)
```


```{r}
# load libraries
library(tidyverse)
library(scran)
library(scales)
library(patchwork)
library(parallel)


# load the plot theme
source("/g/huber/users/hirschmueller/DECODE/MasterThesis/scripts/plot_theme.R")
```


```{r load data}
# from here /g/huber/users/hirschmueller/DECODE/analysis/notch_3prime/scripts/preprocessing_QC.Rmd
raw_sce_notch <- qs::qread("/g/huber/users/hirschmueller/DECODE/analysis/notch_3prime/results/preprocessing/sce_raw.qs")

# from here /g/huber/users/hirschmueller/DECODE/analysis/ctrl_3prime/scripts/preprocessing_QC.Rmd
raw_sce_ctrl <-  qs::qread("/g/huber/users/hirschmueller/DECODE/analysis/ctrl_3prime/results/preprocessing/sce_raw.qs")

raw_sce_combined <- list("ctrl_rep1" = raw_sce_ctrl$TX22, 
                         "ctrl_rep2" = raw_sce_ctrl$TX23,
                         "notch_rep1" = raw_sce_notch$TX22_N, 
                         "notch_rep2" = raw_sce_notch$TX23_N)
```


```{r}
knee_df <- lapply(setNames(names(raw_sce_combined), names(raw_sce_combined)), function(sample) {
  tmp <- raw_sce_combined[[sample]]
  
  knee_df <- DropletUtils::barcodeRanks(tmp) %>%
    data.frame() %>%
    arrange(rank)
})

####################
# ctrl replicate 1 #
####################
knee_plot_ctrl_rep1 <- ggplot(knee_df$ctrl_rep1, aes(x=rank, y=total))+
  geom_line()+
  geom_hline(yintercept = 1000,linetype='dotted', col = 'red', linewidth=1)+
  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
              labels = trans_format("log10", math_format(10^.x))) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
              labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks() +
  theme_Publication() %+%
  theme(plot.title = element_text(size=11, face="bold",hjust = 0.5)) + 
  ylab("Total UMIs") + 
  xlab("Rank") + 
  ggtitle("Control,\nReplicate 1")


####################
# ctrl replicate 2 #
####################
knee_plot_ctrl_rep2 <- ggplot(knee_df$ctrl_rep2, aes(x=rank, y=total))+
  geom_line()+
  geom_hline(yintercept = 750,linetype='dotted', col = 'red', linewidth=1)+
  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
              labels = trans_format("log10", math_format(10^.x))) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
              labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks() +
  theme_Publication() %+%
  theme(plot.title = element_text(size=11, face="bold",hjust = 0.5),
        axis.title.y = element_blank()) + 
  ylab("Total UMIs") + 
  xlab("Rank")+
  ggtitle("Control,\nReplicate 2")


#####################
# notch replicate 1 #
#####################
knee_plot_notch_rep1 <- ggplot(knee_df$notch_rep1, aes(x=rank, y=total))+
  geom_line()+
  geom_hline(yintercept = 750,linetype='dotted', col = 'red', linewidth=1)+
  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
              labels = trans_format("log10", math_format(10^.x))) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
              labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks() +
  theme_Publication() %+%
  theme(plot.title = element_text(size=11, face="bold",hjust = 0.5),
        axis.title.y = element_blank()) + 
  ylab("Total UMIs") + 
  xlab("Rank")+
  ggtitle("Notch,\nReplicate 1")


#####################
# notch replicate 2 #
#####################
knee_plot_notch_rep2 <- ggplot(knee_df$notch_rep2, aes(x=rank, y=total))+
  geom_line()+
  geom_hline(yintercept = 400,linetype='dotted', col = 'red', linewidth=1)+
  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
              labels = trans_format("log10", math_format(10^.x))) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
              labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks() +
  theme_Publication() %+%
  theme(plot.title = element_text(size=11, face="bold",hjust = 0.5),
        axis.title.y = element_blank()) + 
  ylab("Total UMIs") + 
  xlab("Rank")+
  ggtitle("Notch,\nReplicate 2")

knee_plots <- wrap_plots(list(knee_plot_ctrl_rep1, knee_plot_ctrl_rep2,
                              knee_plot_notch_rep1, knee_plot_notch_rep2), 
                         nrow = 1) %>% ggplotify::as.ggplot()

# now remove cells according to the thresholds determined with the knee plots
raw_sce_combined <- lapply(setNames(names(raw_sce_combined), names(raw_sce_combined)), function(sample) {
  tmp <- raw_sce_combined[[sample]]
  tmp$sum = colSums(counts(tmp))

  cutoff <- case_when(
    sample=="ctrl_rep1" ~ 1000,
    sample=="ctrl_rep2" ~ 750,
    sample=="notch_rep1" ~ 750,
    sample=="notch_rep2" ~ 400,
  )
  mask <- tmp$sum > cutoff
  tmp <- tmp[,mask]
  tmp <- logNormCounts(tmp)
  return(tmp)
})





```

```{r}
# load data post qc

# generated here: /g/huber/users/hirschmueller/DECODE/analysis/notch_3prime/scripts/preprocessing_QC.Rmd
qc_sce_notch <- qs::qread("/g/huber/users/hirschmueller/DECODE/analysis/notch_3prime/results/preprocessing/sce_qc_separated_no_batchnorm.qs")

# generated here: /g/huber/users/hirschmueller/DECODE/analysis/ctrl_3prime/scripts/preprocessing_QC.Rmd
qc_sce_ctrl <- qs::qread("/g/huber/users/hirschmueller/DECODE/analysis/ctrl_3prime/results/preprocessing/sce_qc_separated_no_batchnorm.qs")

qc_sce_combined <-  list("ctrl_rep1" = qc_sce_ctrl$TX22, 
                         "ctrl_rep2" = qc_sce_ctrl$TX23,
                         "notch_rep1" = qc_sce_notch$TX22_N, 
                         "notch_rep2" = qc_sce_notch$TX23_N)


mitochondrial_genes <- lapply(setNames(names(raw_sce_combined), names(raw_sce_combined)), function(sample){
  rowData(raw_sce_combined[[sample]])$chromosome_name == "mitochondrion_genome"
})





# calculate qc metrics for the raw data
qc_metrics_raw <- mclapply(setNames(names(raw_sce_combined),names(raw_sce_combined)), function(x){
  tmp <- raw_sce_combined[[x]]
  res <- perCellQCMetrics(logcounts(tmp),  
                          subsets=list(Mito=mitochondrial_genes[[x]])) %>%
    data.frame() %>%
    mutate(sample=x)
  return(res)
}, mc.cores=4) %>% 
  bind_rows() %>% 
  mutate(sample = case_when(
    sample == "ctrl_rep1" ~ "Control\nReplicate 1", 
    sample == "ctrl_rep2" ~ "Control\nReplicate 2", 
    sample == "notch_rep1" ~ "Notch\nReplicate 1", 
    sample == "notch_rep2" ~ "Notch\nReplicate 2", 
  ))

# calculate qc metrics for filtered and normalized data
qc_metrics_post <- mclapply(setNames(names(qc_sce_combined),names(qc_sce_combined)), function(x){
  tmp <- qc_sce_combined[[x]]
  res <- perCellQCMetrics(logcounts(tmp),  subsets=list(Mito=mitochondrial_genes[[x]])) %>%
    data.frame() %>%
    mutate(sample=x)
  return(res)
}, mc.cores=4) %>% 
  bind_rows() %>% 
  mutate(sample = case_when(
    sample == "ctrl_rep1" ~ "Control\nReplicate 1", 
    sample == "ctrl_rep2" ~ "Control\nReplicate 2", 
    sample == "notch_rep1" ~ "Notch\nReplicate 1", 
    sample == "notch_rep2" ~ "Notch\nReplicate 2", 
  ))



####### 
# RAW #
#######
raw_total_plot <- ggplot(qc_metrics_raw, aes(color=sample,x=sample, y=sum)) +
  geom_violin() +
  ylab("Number of UMIs\nper cell") +
  theme_Publication()%+%
  theme(legend.position = "none")+
  ylim(-155,3500)+
  scale_color_manual(values=color_mapping) +
  scale_fill_manual(values=color_mapping) +
  theme(axis.title.x = element_blank())+
  ggtitle("Pre-QC")
  

raw_genes_plot <- ggplot(qc_metrics_raw, aes(color=sample,x=sample, y=detected)) +
  geom_violin() +
  ylab("Number of genes\nper cell") +
  theme_Publication()%+%
  theme(legend.position = "none")+
  scale_color_manual(values=color_mapping) +
  scale_fill_manual(values=color_mapping) +
  theme(axis.title.x = element_blank())+
  ylim(-200,3300)+
  ggtitle("Pre-QC")

raw_mito_plot <- ggplot(qc_metrics_raw, aes(color=sample,x=sample, y=subsets_Mito_percent)) +
  geom_violin() +
  ylab("Mitochondrial\nreads [%]") +
  theme_Publication()%+%
  theme(legend.position = "none")+
  scale_color_manual(values=color_mapping) +
  scale_fill_manual(values=color_mapping) +
  theme(axis.title.x = element_blank())+
  ylim(-5,75)+
  ggtitle("Pre-QC")



######## 
# POST #
########
post_total_plot <-ggplot(qc_metrics_post, aes(color=sample,x=sample, y=sum)) +
  geom_violin() +
  ylab("Number of UMIs per cell") +
  theme_Publication()%+%
  theme(legend.position = "none")+
  ylim(-155,3500)+
  scale_color_manual(values=color_mapping) +
  scale_fill_manual(values=color_mapping) +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_blank())+
  ggtitle("Post-QC")
  

post_genes_plot <- ggplot(qc_metrics_post, aes(color=sample,x=sample, y=detected)) +
  geom_violin() +
  ylab("Number of genes per cell") +
  theme_Publication()%+%
  theme(legend.position = "none")+
  scale_color_manual(values=color_mapping) +
  scale_fill_manual(values=color_mapping) +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_blank())+
  ylim(-200,3300)+
  ggtitle("Post-QC")

post_mito_plot <- ggplot(qc_metrics_post, aes(color=sample,x=sample, y=subsets_Mito_percent)) +
  geom_violin() +
  ylab("Mitochondrial reads [%]") +
  theme_Publication()%+%
  theme(legend.position = "none")+
  scale_color_manual(values=color_mapping) +
  scale_fill_manual(values=color_mapping) +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_blank())+
  ylim(-1,10)+
  ggtitle("Post-QC")


umi_plot <- (raw_total_plot+post_total_plot) %>% ggplotify::as.ggplot()
genes_plot <- (raw_genes_plot+post_genes_plot) %>% ggplotify::as.ggplot()
mito_plot <- (raw_mito_plot+post_mito_plot) %>% ggplotify::as.ggplot()
```


```{r}
### plot cell number before and after
raw_cell<-table(qc_metrics_raw$sample) %>% enframe(name = "sample", value="num_cells") %>% mutate(QC="pre-QC")
post_cell<-table(qc_metrics_post$sample) %>% enframe(name = "sample", value="num_cells") %>% mutate(QC="post-QC")

plot_cell_df <-rbind(raw_cell,post_cell)

cell_number <- ggplot(plot_cell_df,aes(x=sample,y=num_cells,fill=QC))+
  geom_bar(stat="identity",position="identity",width = 0.5)+
  theme_Publication_side_legend() %+%
  theme(axis.title.x = element_blank(),
        legend.title = element_blank(),
        axis.text.x = element_text(size=8.5)) +
  xlab("Experiment")+
  ylab("Number of\ncells")+
  scale_colour_manual(values=c("darkblue", "#B9C6F0")) +
  scale_fill_manual(values=c("darkblue", "#B9C6F0")) +
  scale_alpha_manual(values=c(.3, .8))


```
 

### Combined plot
```{r}
layout = '
AAAAAA
BBBBBB
CCCCCC
DDDDDD
#EEEE#
' 


combined_plot <- wrap_plots(list(knee_plots,
                                 umi_plot,
                                 genes_plot,
                                 mito_plot,
                                 cell_number),
                            ncol = 1) + 
  plot_layout(design=layout) +
  plot_annotation(tag_levels = "A") & 
  theme(plot.tag = element_text(size = 15, face="bold"))

ggsave(filename = "/g/huber/users/hirschmueller/DECODE/MasterThesis/plots/Sup_fig1_QC.png",
       dpi=500, width =8, height = 12.5)

```
 
![](/g/huber/users/hirschmueller/DECODE/MasterThesis/plots/Sup_fig1_QC.png)

```{r, echo=T}
sessionInfo()
```